cmds/receive.c

```diff
@@ -1067,12 +1280,26 @@ static int process_encoded_write(const char *path, const void *data, u64 offset,
 	if (ret < 0)
 		goto out;
 
-	// NOTE: encoded writes guarantee no partial writes,
-	// so we don't need to handle that possibility.
-	ret = pwritev2(rctx->write_fd, iov, 2, offset, RWF_ENCODED);
-	if (ret < 0) {
-		error("encoded_write: writing to %s failed: %m", path);
+	if (encoded_write) {
+		// NOTE: encoded writes guarantee no partial writes,
+		// so we don't need to handle that possibility.
+		ret = pwritev2(rctx->write_fd, iov, 2, offset, RWF_ENCODED);
+		if (ret == -ENOSPC || ret == -EOPNOTSUPP) {
```

Comment:
This comment might've gotten lost before, but

1. This (like most syscalls in libc) returns -1 and sets `errno` on error.
2.  We need to handle `EINVAL` in the same way to handle cases where the sending filesystem has a smaller sectorsize than the receiving filesystem.